pyaedt/generic/general_methods.py

```diff
@@ -40,50 +40,63 @@ def _exception(ex_info, func, args, kwargs, message="Type Error"):
     -------
 
     """
-    print("**************************************************************")
-    print("pyaedt Error on Method {}:  {}. Please Check again".format(func.__name__, message))
-    print("Arguments Provided: ")
-    try:
-
-        if int(sys.version[0]) > 2:
-            args_name = list(OrderedDict.fromkeys(inspect.getfullargspec(func)[0] + list(kwargs.keys())))
-            args_dict = OrderedDict(list(itertools.zip_longest(args_name, args)) + list(kwargs.items()))
-        else:
-            args_name = list(OrderedDict.fromkeys(inspect.getargspec(func)[0] + list(kwargs.keys())))
-            args_dict = OrderedDict(list(itertools.izip(args_name, args)) + list(kwargs.iteritems()))
-
-        for el in args_dict:
-            if el != "self":
-                print("    {} = {} ".format(el, args_dict[el]))
-    except:
-        if len(args) > 1:
-            print(args[1:], kwargs)
-        else:
-            print(kwargs)
+    if  os.getenv('PYAEDT_SCREEN_LOGS','True').lower() in ('true', '1', 't'):
+        print("**************************************************************")
+        print("pyaedt Error on Method {}:  {}. Please Check again".format(func.__name__, message))
+        print("Arguments Provided: ")
+        try:
+
+            if int(sys.version[0]) > 2:
+                args_name = list(OrderedDict.fromkeys(inspect.getfullargspec(func)[0] + list(kwargs.keys())))
+                args_dict = OrderedDict(list(itertools.zip_longest(args_name, args)) + list(kwargs.items()))
+            else:
+                args_name = list(OrderedDict.fromkeys(inspect.getargspec(func)[0] + list(kwargs.keys())))
+                args_dict = OrderedDict(list(itertools.izip(args_name, args)) + list(kwargs.iteritems()))
+
+            for el in args_dict:
+                if el != "self":
+                    print("    {} = {} ".format(el, args_dict[el]))
+        except:
+            if len(args) > 1:
+                print(args[1:], kwargs)
+            else:
+                print(kwargs)
```

Comment:
This section and PYAEDT_DESKTOP_LOGS should use a common method/function and then print the result from that method.

Reply:
I've refactored it